src/app/core/utils/translate/pwa-translate-compiler.ts

```diff
@@ -52,6 +51,13 @@ export class PWATranslateCompiler implements TranslateCompiler {
    */
   private static SIMPLE_VARIABLE_REGEX = /\{\{\s*(\w+)\s*\}\}/g;
 
+  private translate: () => TranslateService;
+
+  constructor(injector: Injector) {
+    // set cyclic dependency
```

Comment:
Maybe this could be better documented to help those who look at this after us. Why is once() used here? It's not the standard way to inject dependencies, after all.

I propose the following comment: "cache translateService reference to avoid repeated calls to injector" or something similar. What do you think, @dhhyi ?

Reply:
Go ahead! 😃